src/nvim/api/vim.c

```diff
@@ -696,7 +696,15 @@ Object nvim_get_option_value(String name, Dict(option) *opts, Error *err)
     rv = INTEGER_OBJ(numval);
     break;
   case 2:
-    rv = BOOLEAN_OBJ(!!numval);
+    switch (numval) {
+      case 0:
+      case 1:
+        rv = BOOLEAN_OBJ(numval);
+        break;
+      default:
+        rv = INTEGER_OBJ(numval);
```

Comment:
I'd suggest using `NIL` to represent unset `autoread` value.

Reply:
`-1` is used in `do_set`:

```vim
:set autoread
:setl autoread<
:echo &l:autoread
-1
```

Shouldn't we try and keep the API version congruent with the Vim script version?

Reply:
what congruence? `&l:autoread` **always** returns an integer value, never a bool, so already using booleans breaks that rule in that case. vim script cannot change it to neither `v:true` / `v:false` nor `v:null` due to backwards compatibility, while I think API functions should use API types whenever possible/reasonable.

Reply:
Ok done.